internal/ingress/controller/nginx.go

```diff
@@ -660,16 +672,64 @@ func (n *NGINXController) OnUpdate(ingressCfg ingress.Configuration) error {
 		}
 	}
 
-	err = ioutil.WriteFile(cfgPath, content, 0644)
-	if err != nil {
-		return err
-	}
+	if n.cfg.DynamicReload {
+		src, _ := ioutil.ReadFile(cfgPath)
+		if !bytes.Equal(src, content) {
+			err = ioutil.WriteFile(cfgPath, content, 0644)
+			if err != nil {
+				return err
+			}
 
-	o, err := exec.Command(n.binary, "-s", "reload", "-c", cfgPath).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("%v\n%v", err, string(o))
-	}
+			o, err := exec.Command(n.binary, "-s", "reload", "-c", cfgPath).CombinedOutput()
+			if err != nil {
+				return fmt.Errorf("%v\n%v", err, string(o))
+			}
+		} else {
+			glog.Infof("NGINX reload not needed, executing live update only\n")
```

Comment:
I might be reading the code wrong, but why does live updated needed in this case? It seems like the existing configuration is identical to the new one(referring to line 677).

Similarly if we are reloading at 683, why don't we break? What's the point of continuing with live update after reload?

Reply:
If I'm right, there's the case where config file does not change, but live update needed to include new backends.